app/models/miq_ae_datastore.rb

```diff
@@ -170,17 +170,15 @@ def self.reset_to_defaults
   end
 
   def self.reset_domains_from_legacy_directory
-    domain_files = DATASTORE_DIRECTORY.join('*', MiqAeDomain::DOMAIN_YAML_FILENAME)
-    Dir.glob(domain_files).each do |domain_file|
-      domain_name = File.basename(File.dirname(domain_file))
-      reset_domain(DATASTORE_DIRECTORY, domain_name, Tenant.root_tenant)
-    end
+    legacy_domain_names.each { |domain_name| reset_domain(DATASTORE_DIRECTORY, domain_name, Tenant.root_tenant) }
   end
 
   def self.reset_domains_from_vmdb_plugins
-    datastores_for_reset.each do |domain|
-      reset_domain(domain.datastores_path.to_s, domain.name, Tenant.root_tenant)
-    end
+    plugin_domain_names.each { |domain| reset_domain(domain.datastores_path.to_s, domain.name, Tenant.root_tenant) }
+  end
+
+  def self.domain_names
+    plugin_domain_names + legacy_domain_names
   end
```

Comment:
I think this needs to be taken one step further and the` default_domain_names` method needs to change (line 192).  That method is the one that defines the "system" domains for the UI, but it is ignoring the legacy domains.  The refactoring here should mostly help with getting those legacy domains into that list.